internal/proto/volumegroup.proto

```diff
@@ -0,0 +1,182 @@
+syntax = "proto3";
+package proto;
+
+option go_package = "github.com/csi-addons/kubernetes-csi-addons/internal/proto";
+
+// Controller holds the RPC methods for volumeGroup and all the methods it
+// exposes should be idempotent.
+service Controller {
+  // CreateVolumeGroup RPC call to create a volume group.
+  rpc CreateVolumeGroup(CreateVolumeGroupRequest)
+      returns (CreateVolumeGroupResponse) {}
+  // ModifyVolumeGroupMembership RPC call to modify a volume group.
+  rpc ModifyVolumeGroupMembership(ModifyVolumeGroupMembershipRequest)
+      returns (ModifyVolumeGroupMembershipResponse) {}
+  // DeleteVolumeGroup RPC call to delete a volume group.
+  rpc DeleteVolumeGroup(DeleteVolumeGroupRequest)
+      returns (DeleteVolumeGroupResponse) {}
+  // ListVolumeGroups RPC call to list volume groups.
+  // rpc ListVolumeGroups(ListVolumeGroupsRequest)
+  //    returns (ListVolumeGroupsResponse) {}
+  // CreateVolumeGroup RPC call to get a volume group.
+  rpc ControllerGetVolumeGroup(ControllerGetVolumeGroupRequest)
+      returns (ControllerGetVolumeGroupResponse) {}
+}
+// CreateVolumeGroupRequest holds the required information to
+// create a volume group
+message CreateVolumeGroupRequest {
+  // suggested name for volume group (required for idempotency)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // params passed to the plugin to create the volume group.
+  // This field is OPTIONAL.
+  map<string, string> parameters = 2;
+
+  // Secrets required by plugin to complete volume group creation
+  // request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 3;
```

Comment:
secrets should not be sent between controller and sidecar. The sidecar should be able to resolve the secrets based on namespace/secret-name.